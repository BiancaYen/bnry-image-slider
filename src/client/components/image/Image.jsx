// Image

import PropTypes from 'prop-types';
import React, { Fragment, useState } from 'react';

// Components
import ImagePlaceholder from './placeholder/ImagePlaceholder';

// Styles
import { ImageTag } from './styles';

// Default Props
const defaultProps = {
    title: 'No title provided'
};

// Prop Types
const propTypes = {
    source: PropTypes.string.isRequired,
    title: PropTypes.string
};

const Image = ({ source, title }) => {
    const [error, setError] = useState(false);

    const handleImageError = () => {
        if (!error) {
            setError(true);
        }
    };

    return (
        <Fragment>
            {
                !error
                    ? (
                        <ImageTag
                            alt={title}
                            src={source}
                            onError={handleImageError}
                        />
                    )
                    : <ImagePlaceholder title={title} />
            }
        </Fragment>
    );
};

Image.defaultProps = defaultProps;
Image.propTypes = propTypes;

export default Image;
