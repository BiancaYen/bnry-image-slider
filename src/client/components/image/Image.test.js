// Image Test

import { mount, shallow } from 'enzyme';
import React from 'react';

// Components
import Image from './Image';

// Constants
const source = 'https://images.pexels.com/photos/1591447/pexels-photo-1591447.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500';
const title = 'Title';

describe('Image', () => {
    it('renders with all required prop types', () => {
        const component = shallow(
            <Image source={source} />
        );

        expect(component).toMatchSnapshot();
    });

    it('asserts that prop types are correct', () => {
        const component = mount(
            <Image
                source={source}
                title={title}
            />
        );

        expect(component.find('img').prop('src')).toEqual(source);
        expect(component.find('img').prop('alt')).toEqual(title);
    });
});
