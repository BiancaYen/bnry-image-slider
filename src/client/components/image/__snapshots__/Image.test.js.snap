// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders with all required prop types 1`] = `
<Fragment>
  <ImageTag
    alt="No title provided"
    onError={[Function]}
    src="https://images.pexels.com/photos/1591447/pexels-photo-1591447.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
  />
</Fragment>
`;
